diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/TrailingCommaRule.kt

```diff
@@ -0,0 +1,114 @@
+ * [10] Type arguments
+ * [11] Type parameters
+ * [12] Destructuring declarations
+ */
+class TrailingCommaRule(private val configRules: List<RulesConfig>) : Rule("trailing-comma") {
+    private var isFixMode: Boolean = false
+    private val configuration by lazy {
+        TrailingCommaConfiguration(
+            this.configRules.getRuleConfig(TRAILING_COMMA)?.configuration ?: emptyMap()
+        )
+    }
+    private lateinit var emitWarn: EmitType
+
+    override fun visit(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: EmitType
+    ) {
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        val commonConfig by configRules.getCommonConfiguration()
+        if (commonConfig.kotlinVersion >= ktVersion) {
+            val (type, config) = when (node.elementType) {
+                VALUE_ARGUMENT_LIST -> Pair(VALUE_ARGUMENT, configuration.getParam("valueArgument"))
+                VALUE_PARAMETER_LIST -> Pair(VALUE_PARAMETER, configuration.getParam("valueParameter"))
+                INDICES -> Pair(REFERENCE_EXPRESSION, configuration.getParam("referenceExpression"))
+                WHEN_ENTRY -> Pair(WHEN_CONDITION_WITH_EXPRESSION, configuration.getParam("whenConditions"))
+                COLLECTION_LITERAL_EXPRESSION -> Pair(STRING_TEMPLATE, configuration.getParam("collectionLiteral"))
+                TYPE_ARGUMENT_LIST -> Pair(TYPE_PROJECTION, configuration.getParam("typeArgument"))
+                TYPE_PARAMETER_LIST -> Pair(TYPE_PARAMETER, configuration.getParam("typeParameter"))
+                DESTRUCTURING_DECLARATION -> Pair(
+                    DESTRUCTURING_DECLARATION_ENTRY,
+                    configuration.getParam("destructuringDeclaration")
+                )
+                else -> return
+            }
+            val astNode = node
+                .children()
+                .toList()
+                .ifNotEmpty { this.lastOrNull { it.elementType == type } }
+            astNode?.checkTrailingComma(config)
+        }
+    }
+
+    private fun ASTNode.checkTrailingComma(config: Boolean) {
+        val isNextComma = this.siblings(true).map { it.elementType }.contains(COMMA)
+        if (!isNextComma && config) {
+            // we should write type of node in warning, to make it easier for user to find the parameter
+            TRAILING_COMMA.warnAndFix(configRules, emitWarn, isFixMode, "after ${this.elementType}: ${this.text}", this.startOffset, this) {
+                val parent = this.treeParent
+                parent.addChild(LeafPsiElement(COMMA, ","), this.treeNext)
+            }
+        }
+    }
+
+    /**
+     * Configuration for trailing comma
+     */
+    class TrailingCommaConfiguration(config: Map<String, String>) : RuleConfiguration(config) {
+        /**
+         * @param name parameters name
+         * @return param based on its name
+         */
+        fun getParam(name: String) = config[name]?.toBoolean() ?: false
```

Comment:
Maybe, make `true` default? If user enables this rule without configuration, then all cases will be active. Then they can suppress individual settings for their liking.

Reply:
It seems to me better to leave everything off, because the relatively new feature and complex configuration.

Reply:
Then maybe log a warning if the rule is enabled, but all options are not set? Like `You have enabled TRAILING_COMMA, but rule will remain inactive until you explicitly set configuration options. See [available-rules.md] for possible configuration options.`

Reply:
Yes you are right. Fixed